### PR TITLE
SALTO-6605 SOAP WSDL version config

### PR DESCRIPTION
**Pull request by dantal4**
Allow users to config the SOAP WSDL version used by `SoapClient` - using the `suiteAppClient.wsdlVersion` adapter config.

---

_Additional context for reviewer_

---
_Release Notes_: 
Netsuite Adapter:
- Allow users to config the SOAP WSDL version used by `SoapClient` - using the `suiteAppClient.wsdlVersion` adapter config

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/client/suiteapp_client/soap_client/types.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@
  *
  * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
  */
+
+export const SUPPORTED_WSDL_VERSIONS = ['2020_2', '2023_1', '2023_2', '2024_1'] as const
```
Comment (omrilit) [inline packages/netsuite-adapter/src/client/suiteapp_client/soap_client/types.ts:9]: Do you think that we should allow also future versions by having a regex validation in the adapter_creator?
Just so we will have the option to upgrade to newer versions without code changes.
This is probably not needed for now and we can consider doing it in the future if this is something you'd prefer

File: packages/netsuite-adapter/src/config/validations.ts [hunk 2]
```diff
@@ -72,6 +73,26 @@ function validateNumber(value: unknown, configPath: string | string[]): asserts
   }
 }
 
+function validateEnumValue(
+  value: unknown,
+  enumValues: ReadonlyArray<string>,
+  configPath: string | string[],
+): asserts value is string | undefined {
+  if (value === undefined) {
```
Comment (omrilit) [inline packages/netsuite-adapter/src/config/validations.ts:81]: You already check outside whether the value is undefined, maybe one of them is redundant?

**Review by omrilit — APPROVED** [review]
NICE!

File: packages/netsuite-adapter/src/client/suiteapp_client/soap_client/types.ts (hunk 1 above)
Comment (dantal4) [reply packages/netsuite-adapter/src/client/suiteapp_client/soap_client/types.ts:9]: if a user will use a version that match the regex and doesn't exist, they'll get an exception from SOAP, so I think that I prefer that we'll manage the allowed versions.

**Pull request merged**